### PR TITLE
feat: Move get asset size logic to use case in Kalium

### PR DESCRIPTION
**Pull request by gongracr**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
Currently, the logic regarding what was the maximum size allowed for assets to be sent was only used on the `MessageComposerViewModel`. Since now we are introducing the import external media assets feature, we also needed to access this logic, and therefore the obvious choice was to move this logic into Kalium to be consumed as a use case.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAssetSizeLimitUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+fun interface GetAssetSizeLimitUseCase {
+    /**
+     * Returns the maximum size of an asset that can be uploaded to the Wire backend.
+     * @param isImage whether the asset to upload is an image or not
+     */
+    suspend operator fun invoke(isImage: Boolean): Long
+}
+
+internal class GetAssetSizeLimitUseCaseImpl internal constructor(
+    private val isSelfATeamMember: IsSelfATeamMemberUseCase
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAssetSizeLimitUseCase.kt:14]: I think we can inject here the `Dispatcher` and switching the execution here, so we don't have to do it on clients :D

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsSelfATeamMemberUseCase.kt [hunk 2]
```diff
@@ -1,33 +1,19 @@
...
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program. If not, see http://www.gnu.org/licenses/.
- */
-
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsSelfATeamMemberUseCase.kt:18]: I think we can maintain the copyright notice

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsSelfATeamMemberUseCase.kt [hunk 3]
```diff
@@ -1,33 +1,19 @@
...
+    suspend operator fun invoke(): Boolean
+}
+
 /**
- * Return if self user is part of a team or noz
+ * Return if self user is part of a team or not
  */
-class IsSelfATeamMemberUseCase internal constructor(
+internal class IsSelfATeamMemberUseCaseImpl internal constructor(
     private val selfTeamIdProvider: SelfTeamIdProvider
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsSelfATeamMemberUseCase.kt:14]: Same here, we can inject the dispatchers and gain some standardization on our Use Cases

**Review by yamilmedina — COMMENTED** [review]
Great idea with the refactor needed also for the import feature.
Just some small comments 🪁

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAssetSizeLimitUseCase.kt (hunk 1 above)
Comment (gongracr) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAssetSizeLimitUseCase.kt:14]: Not sure if we should switch the context on the use case level or rather on the repository, data layer level. Maybe something to discuss on next collective?

**Pull request merged**